src/services/image.service.ts

```diff
@@ -0,0 +1,125 @@
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from "fs";
+import path, { join, resolve } from "path";
+import { v4 } from "uuid";
+import { HttpError } from "../types/custom.error";
+import * as fs from "fs";
+
+export interface FileMetadata {
+  id: string;
+  filePath: string;
+  filename: string;
+  originalName: string;
+}
+
+export type CreateImage = Omit<
+  {
+    id: string;
+    filename: string;
+    originalName: string;
+    path: string;
+  },
+  "id"
+>;
+
+class ImageService {
+  private defaultPath = resolve("./covers");
+  private metadataFilePath = resolve(__dirname, "../database/metadata.json");
+
+  private validateMimeType(mimeType: string) {
+    if (!mimeType.includes("image")) {
+      throw new HttpError("File type is not supported", 400);
+    }
+  }
+
+  private async saveFile(
+    file: Express.Multer.File,
+    dir: string,
+    filePath: string
+  ) {
+    try {
+      if (!existsSync(dir)) {
+        mkdirSync(dir, { recursive: true });
+      }
+      writeFileSync(filePath, file.buffer);
+    } catch (error) {
+      throw new HttpError("Error trying to save a file", 500);
+    }
+  }
+
+  private saveMetadata(metadata: FileMetadata[]) {
+    writeFileSync(this.metadataFilePath, JSON.stringify(metadata));
+  }
+
+  private readMetadata(): FileMetadata[] {
+    try {
```

Comment:
I have a question, I don't quite understand what you need to change since to access the image there is the findImage() method there it receives the id:uuid, this is searched in the json to avoid the duplicate since uuid cannot be compared because They are different, so in the json I keep the originalName and that is the one that avoids the duplicate.

In the findImage() it also looks for the id that it receives in the json and returns the name of the image and it is sent along with the path where the images are and in the controller it already sends the image with that information.

So maybe I need a more detailed explanation of what I should do because I'm not really getting it.